assignments/prototypes.js

```diff
@@ -1,13 +1,42 @@
 /*
   Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
 
-  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.
 
   At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
-  
+
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+  function GameObject(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+    this.name = attributes.name;
+  }
+
+  GameObject.prototype.destroy = function (name){
+    return `${this.name} was removed from the game.`;
+  }
+
+
+    function CharacterStats(attributes) {
+      GameObject.call(this, attributes);
+      this.healthPoints = attributes.healthPoints;
+    }
+
+    CharacterStats.prototype.takeDamage = function (name){
+      return `${this.name} took damage.`;
+    }
+
+    function Humanoid(attributes) {
+      CharacterStats.call(this, attributes);
+      this.team = attributes.team;
+      this.weapons = attributes.weapons;
+      this.language = attributes.language;
+    }
+
```

Comment:
This is another place to set up inheritance between the above block of code, and the one directly below. Consider:
Humanoid.protoype = Object.create(CharacterStats.prototype);